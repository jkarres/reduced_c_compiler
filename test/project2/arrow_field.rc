structdef ST {
  int i;
  float f;
  bool b;
  function : void print() {
    cout << "<ST";
    cout << " i:" << this.i;
    cout << " f:" << this.f;
    cout << " b:" << this.b;
    cout << ">" << endl;
  }
};


ST gs;

function : void main() {
  cout << "ST gs in bss." << endl;

  cout << "Creating ST ls on the stack." << endl;
  ST ls;
  cout << endl;

  cout << "Creating ST * ps on the stack, pointing to gs." << endl;
  ST * ps = &gs;

  cout << "*ps:";
  ps->print();
  cout << endl;

  cout << "Setting ps->i to 123, ps->f to 987, and ps->b to true." << endl;
  ps->i = 123;
  ps->f = 987;
  ps->b = true;
  cout << endl;

  cout << "*ps:";
  ps->print();
  cout << endl;

  cout << "gs:";
  gs.print();
  cout << endl;


  cout << "Setting ps to &ls." << endl;
  ps = &ls;


  cout << "Setting ps->i to 456, ps->f to 654, and ps->b to false." << endl;
  ps->i = 456;
  ps->f = 654;
  ps->b = false;

  cout << "*ps:";
  ps->print();
  cout << "ls:";
  ls.print();
  cout << endl;

  cout << "*ps = *&gs" << endl;
  *ps = *&gs;


  cout << "*ps:";
  ps->print();
  cout << endl;

  cout << "ls:";
  ls.print();

}
