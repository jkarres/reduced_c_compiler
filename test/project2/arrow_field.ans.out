ST gs in bss.
Creating ST ls on the stack.

Creating ST * ps on the stack, pointing to gs.
*ps:<ST i:0 f:0.00 b:false>

Setting ps->i to 123, ps->f to 987, and ps->b to true.

*ps:<ST i:123 f:987.00 b:true>

gs:<ST i:123 f:987.00 b:true>

Setting ps to &ls.
Setting ps->i to 456, ps->f to 654, and ps->b to false.
*ps:<ST i:456 f:654.00 b:false>
ls:<ST i:456 f:654.00 b:false>

*ps = *&gs
*ps:<ST i:123 f:987.00 b:true>

ls:<ST i:123 f:987.00 b:true>
